### PR TITLE
Create frontend model from capi sponsorship model

### PR DESCRIPTION
**Pull request by kelvin-chappell**
This integrates the new sponsorship data from Capi responses into the frontend content model.  
There's no change in what's rendered yet so this bit doesn't need to be behind a feature switch.

/cc @steppenwells

### CODE REVIEW

File: common/app/model/Tag.scala [hunk 1]
```diff
@@ -114,6 +117,88 @@ case class Reference(
...
+  override val name: String = "foundation"
+}
+
+case object PaidContent extends SponsorshipType {
+  override val name: String = "paid-content"
+}
+
+object SponsorshipType {
+
+  def make(name: String): SponsorshipType = name match {
```
Comment (janua) [inline common/app/model/Tag.scala]: Is this function intentionally partial?

Comment (kelvin-chappell) [inline common/app/model/Tag.scala]: No.  I'll complete it.

Comment (janua) [inline common/app/model/Tag.scala]: :clap:

**Comment by kelvin-chappell** [comment]
@janua Is that better?

**Comment by janua** [comment]
LGTM :+1:

**Pull request merged**